src/pke/lib/scheme/ckksrns/ckksrns-schemeswitching.cpp

```diff
@@ -0,0 +1,2152 @@
+#endif
+
+    auto ccCKKS = privateKey->GetCryptoContext();
+
+    uint32_t M       = ccCKKS->GetCyclotomicOrder();
+    uint32_t slots   = m_numSlotsCKKS;
+    uint32_t n       = lwesk->GetElement().GetLength();
+    uint32_t ringDim = ccCKKS->GetRingDimension();
+
+    // Intermediate cryptocontext for CKKS to FHEW
+    auto keys2 = m_ccKS->KeyGen();
+
+    Plaintext ptxtZeroKS = m_ccKS->MakeCKKSPackedPlaintext(std::vector<double>{0.0}, 1, 0, nullptr, slots);
+    m_ctxtKS             = m_ccKS->Encrypt(keys2.publicKey, ptxtZeroKS);
+
+    // Compute switching key between RLWE and LWE via the intermediate cryptocontext, keep it in RLWE form
+    m_CKKStoFHEWswk = switchingKeyGenRLWEcc(keys2.secretKey, privateKey, lwesk);
+
+    auto skLWEElements = lwesk->GetElement();
+    std::vector<std::complex<double>> skLWEDouble(n);
+    for (uint32_t i = 0; i < n; i++) {
+        auto tmp = skLWEElements[i].ConvertToDouble();
+        if (tmp == lwesk->GetModulus().ConvertToInt() - 1)
+            tmp = -1;
+        skLWEDouble[i] = std::complex<double>(tmp, 0);
+    }
+
+    // Check encoding and specify the number of slots, otherwise, if batchsize is set and is smaller, it will throw an error.
+    Plaintext skLWEPlainswk;
+    if (cryptoParams->GetScalingTechnique() == FLEXIBLEAUTOEXT)
+        skLWEPlainswk = ccCKKS->MakeCKKSPackedPlaintext(Fill(skLWEDouble, ringDim / 2), 1, BASE_NUM_LEVELS_TO_DROP,
+                                                        nullptr, ringDim / 2);
+    else
+        skLWEPlainswk = ccCKKS->MakeCKKSPackedPlaintext(Fill(skLWEDouble, ringDim / 2), 1, 0, nullptr, ringDim / 2);
+
+    m_FHEWtoCKKSswk = ccCKKS->Encrypt(publicKey, skLWEPlainswk);
+
+    // Compute automorphism keys
+    /* CKKS to FHEW */
+    if (dim1CF == 0)
+        dim1CF = getRatioBSGSLT(static_cast<double>(slots));
+    m_dim1CF = dim1CF;
+    m_LCF    = LCF;
+
+    // Compute indices for rotations for slotToCoeff transform
+    std::vector<int32_t> indexRotationS2C = FindLTRotationIndicesSS(m_dim1CF, M, slots);
+    indexRotationS2C.emplace_back(static_cast<int>(slots));
+
+    // Compute indices for rotations for sparse packing
+    for (uint32_t i = 1; i < ccCKKS->GetRingDimension() / 2; i *= 2) {
+        indexRotationS2C.emplace_back(static_cast<int>(i));
+        if (i <= slots)
+            indexRotationS2C.emplace_back(-static_cast<int>(i));
+    }
+
+    /* FHEW to CKKS */
+    if (dim1FC == 0)
+        dim1FC = getRatioBSGSLT(static_cast<double>(n));  // This picks the ratio for baby-step giant-step
+    m_dim1FC = dim1FC;
+    m_LFC    = LFC;
+
+    // Compute indices for rotations for homomorphic decryption in CKKS
+    std::vector<int32_t> indexRotationHomDec = FindLTRotationIndicesSS(m_dim1FC, M, n);
+
+    // Compute indices for rotations to bring back the final CKKS ciphertext encoding to slots
+    for (uint32_t j = 1; j < ringDim / (2 * slots); j <<= 1) {
```

Comment:
indexRotationHomDec should be resized before it gets more elements (call reserve())